### PR TITLE
Intel compiler fixes

### PR DESCRIPTION
**Pull request by dschwen**
Fixes a tiny issue in the new grain tracker and a bunch of things introduced in #6802 

Yeah... it's not pretty.

### CODE REVIEW

File: modules/phase_field/include/postprocessors/GrainTracker.h [hunk 1]
```diff
@@ -138,7 +138,7 @@ class GrainTracker : public FeatureFloodCount, public GrainTrackerInterface
   const unsigned int _halo_level;
 
   /// Depth of renumbing recursion (a depth of zero means no recursion)
-  const unsigned int _max_renumbering_recursion = 2;
+  static const unsigned int _max_renumbering_recursion = 2;
```
Comment (permcody) [inline modules/phase_field/include/postprocessors/GrainTracker.h]: Cool thanks for catching this. I might make this into a parameter eventually.

**Comment by permcody** [comment]
👍

**Comment by tophmatthews** [comment]
+1 this fix!

This caused some issues with our intel compiler. Do you guys test against an intel compiler?

**Comment by dschwen** [comment]
Yes, but not on pull requests, only for devel->master. So only master is guaranteed to work with intel compilers.

**Comment by tophmatthews** [comment]
We were on master branch and were failing our builds before this latest fix?

**Comment by permcody** [comment]
We only test *on* the master branch which means we don't know it's broken
until it's already in there. We are going to change that starting next week
(Yes Brian and Jason, you heard that correctly). We've been waiting until
we had additional testing hardware which just went online today. We'll add
additional jobs to catch more of these problems before they show up in
master. Sorry for the trouble.

Cody

**Comment by tophmatthews** [comment]
Great to hear it, sounds like bad timing on our part, thanks guys!

**Comment by dschwen** [comment]
Ha, well, I guess my answer was ahead of its time then :-D

**Pull request merged**